docs: add documentation for web3 names (#62)
KILTprotocol/docs: docs/sdk/0-core-feature/1_web3names.md

---
id: web3names
title: Web3 names
---
import CodeBlock from '@theme/CodeBlock';
import Example1 from '!!raw-loader!../../../code-examples/core_features/web3names/1_web3name.ts';
import Example2 from '!!raw-loader!../../../code-examples/core_features/web3names/2_web3name.ts';
import Example3 from '!!raw-loader!../../../code-examples/core_features/web3names/3_web3name.ts';

Web3 names are user-friendly aliases for a KILT DID.

> They serve the same purpose that domain names do for IP addresses: who knows what are all the IP addresses under the `google.com` domain name? 🤷🏽‍♀️

Reviewer: I would maybe keep this in the singular since IP Address = DID and W3N = Domain. If we have many IPs for a Domain, can we have many DIDs for a W3N too? It kinda suggests that by having the plural there.

```suggestion
They serve the same purpose that domain names do for IP addresses: who knows the IP address under the `kilt.io` domain name? 🤷🏽‍♀️
```

Author: Better example 😁 https://github.com/KILTprotocol/docs/pull/62/commits/febafddcc2aa84f5786361dc1a3b96f91594852f

> Each Web3 name is globally unique, and is composed of a sequence of a minimum of 3 to a maximum of 32 characters taken from a specific character set to enhance human redability and reduce the changes of "two Web3 names looking the same, despite being different".

Reviewer 2: This Web3 name is unique for the KILT Blockchain? Someone can make a new chain and create a new Web3 name.

Author: https://github.com/KILTprotocol/docs/pull/62/commits/c38463063be852e475c15251f6784f16554fe4be

---
id: web3names
title: Web3 names
---
import CodeBlock from '@theme/CodeBlock';
import Example1 from '!!raw-loader!../../../code-examples/core_features/web3names/1_web3name.ts';
import Example2 from '!!raw-loader!../../../code-examples/core_features/web3names/2_web3name.ts';
import Example3 from '!!raw-loader!../../../code-examples/core_features/web3names/3_web3name.ts';
import Example4 from '!!raw-loader!../../../code-examples/core_features/web3names/4_web3name.ts';

Web3 names are user-friendly aliases for a KILT DID.
They serve the same purpose that domain names do for IP addresses: who knows the IP address under the `kilt.io` domain name? 🤷🏽‍♀️

> Each Web3 name is globally unique within the KILT space, and is composed of a sequence of a minimum of 3 to a maximum of 32 characters taken from a specific character set to enhance human redability and reduce the changes of "two Web3 names looking the same, despite being different".

Reviewer 3: ```suggestion
Each Web3 name is globally unique within the KILT space, and is composed of a sequence of a minimum of 3 to a maximum of 32 characters taken from a specific character set to enhance human redability and reduce the chances of "two Web3 names looking the same, despite being different".
```

Author: Good catch! https://github.com/KILTprotocol/docs/pull/62/commits/708f5a8c81bf88d3211855650d5c78d3a39de1c1
